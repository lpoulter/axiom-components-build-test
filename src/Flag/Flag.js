import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Image from '../Image/Image';
import { flags } from '@brandwatch/axiom-materials';

const alpha2Aliases = {
  uk: 'gb',
};

export default class Flag extends Component {
  static propTypes = {
    /** Alpha2 country code */
    alpha2: PropTypes.oneOf([
      // Official
      'ad', 'ae', 'af', 'ag', 'ai', 'al', 'am', 'ao', 'aq', 'ar', 'as', 'at', 'au', 'aw', 'ax', 'az', 'ba', 'bb', 'bd', 'be', 'bf', 'bg', 'bh', 'bi', 'bj', 'bl', 'bm', 'bn', 'bo', 'bq', 'br', 'bs', 'bt', 'bv', 'bw', 'by', 'bz', 'ca', 'cc', 'cd', 'cf', 'cg', 'ch', 'ci', 'ck', 'cl', 'cm', 'cn', 'co', 'cr', 'cu', 'cv', 'cw', 'cx', 'cy', 'cz', 'de', 'dj', 'dk', 'dm', 'do', 'dz', 'ec', 'ee', 'eg', 'eh', 'er', 'es', 'et', 'fi', 'fj', 'fk', 'fm', 'fo', 'fr', 'ga', 'gb', 'gd', 'ge', 'gf', 'gg', 'gh', 'gi', 'gl', 'gm', 'gn', 'gp', 'gq', 'gr', 'gs', 'gt', 'gu', 'gw', 'gy', 'hk', 'hm', 'hn', 'hr', 'ht', 'hu', 'id', 'ie', 'il', 'im', 'in', 'io', 'iq', 'ir', 'is', 'it', 'je', 'jm', 'jo', 'jp', 'ke', 'kg', 'kh', 'ki', 'km', 'kn', 'kp', 'kr', 'kw', 'ky', 'kz', 'la', 'lb', 'lc', 'li', 'lk', 'lr', 'ls', 'lt', 'lu', 'lv', 'ly', 'ma', 'mc', 'md', 'me', 'mf', 'mg', 'mh', 'mk', 'ml', 'mm', 'mn', 'mo', 'mp', 'mq', 'mr', 'ms', 'mt', 'mu', 'mv', 'mw', 'mx', 'my', 'mz', 'na', 'nc', 'ne', 'nf', 'ng', 'ni', 'nl', 'no', 'np', 'nr', 'nu', 'nz', 'om', 'pa', 'pe', 'pf', 'pg', 'ph', 'pk', 'pl', 'pm', 'pn', 'pr', 'ps', 'pt', 'pw', 'py', 'qa', 're', 'ro', 'rs', 'ru', 'rw', 'sa', 'sb', 'sc', 'sd', 'se', 'sg', 'sh', 'si', 'sj', 'sk', 'sl', 'sm', 'sn', 'so', 'sr', 'ss', 'st', 'sv', 'sx', 'sy', 'sz', 'tc', 'td', 'tf', 'tg', 'th', 'tj', 'tk', 'tl', 'tm', 'tn', 'to', 'tr', 'tt', 'tv', 'tw', 'tz', 'ua', 'ug', 'um', 'us', 'uy', 'uz', 'va', 'vc', 've', 'vg', 'vi', 'vn', 'vu', 'wf', 'ws', 'ye', 'yt', 'za', 'zm', 'zw',
      // Unofficial
      'eu', 'gb-eng', 'gb-nir', 'gb-sct', 'gb-wls', 'un', 'uk',
    ]),
    /** Assets URL */
    assetsUrl: PropTypes.string,
    /** Fallback content when the image fails to load */
    children: PropTypes.node,
    /** Size of the Flag */
    size: PropTypes.string.isRequired,
  };

  static defaultProps = {
    assetsUrl: 'https://axiom.brandwatch.com/assets/',
  };

  render() {
    const {
      alpha2,
      assetsUrl,
      children,
      size,
      ...rest
    } = this.props;

    const flag = (alpha2Aliases[alpha2] ? flags[alpha2Aliases[alpha2]] : flags[alpha2]) || {};
    const src = `${assetsUrl}${flag.src}`;

    return (
      <Image { ...rest }
          height={ size }
          shape="circle"
          src={ src }
          width={ size }>
        { children }
      </Image>
    );
  }
}
