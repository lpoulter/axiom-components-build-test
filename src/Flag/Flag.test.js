import React from 'react';
import renderer from 'react-test-renderer';
import Flag from './Flag';

const getComponent = (props = {}) =>
  renderer.create(
    <Flag { ...props } />
  );

describe('Flag', () => {
  it('renders with custom size', () => {
    const component = getComponent({ alpha2: 'gb', size: '12rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
