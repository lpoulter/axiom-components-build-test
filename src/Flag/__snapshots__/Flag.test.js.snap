// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag renders with custom size 1`] = `
<img
  className="ax-image ax-image--circle ax-space--x4"
  src="https://axiom.brandwatch.com/assets/flags/gb.svg"
  style={
    Object {
      "backgroundColor": undefined,
      "height": "12rem",
      "maxWidth": "100%",
      "width": "12rem",
    }
  }
/>
`;
